backend/src/data/repositories/course/course.repository.ts

```diff
@@ -77,7 +78,7 @@ class Course {
     const { results, total } = await this.#CourseModel
       .query()
       .withGraphJoined('category')
-      .orderBy('courseCategoryId', SortOrder.DESC)
+      .orderByRaw(`course_category_id ${SortOrder.ASC} NULLS FIRST`)
```

Comment:
```suggestion
      .orderByRaw(`:blah: ${SortOrder.ASC} NULLS FIRST`, {
        'blah': 'courseCategoryId',
      })
```
bindings can handle this case